core-services/src/main/java/org/zalando/nakadi/service/StorageService.java

```diff
@@ -24,75 +23,58 @@
 
 import javax.annotation.PostConstruct;
-import javax.annotation.PreDestroy;
 import java.io.IOException;
 import java.util.List;
 import java.util.Optional;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 
-@Service
+@Service("storageService")
 public class StorageService {
 
-    public static final String ZK_TIMELINES_DEFAULT_STORAGE = "/nakadi/timelines/default_storage";
     private static final Logger LOG = LoggerFactory.getLogger(StorageService.class);
     private final ObjectMapper objectMapper;
     private final StorageDbRepository storageDbRepository;
-    private final DefaultStorage defaultStorage;
-    private final CuratorFramework curator;
     private final FeatureToggleService featureToggleService;
     private final NakadiAuditLogPublisher auditLogPublisher;
-    private final ScheduledExecutorService zkNotificationThread;
+    private final TransactionTemplate transactionTemplate;
+    private final Environment environment;
 
     @Autowired
     public StorageService(final ObjectMapper objectMapper,
                           final StorageDbRepository storageDbRepository,
-                          @Qualifier("default_storage") final DefaultStorage defaultStorage,
-                          final ZooKeeperHolder zooKeeperHolder,
                           final FeatureToggleService featureToggleService,
-                          final NakadiAuditLogPublisher auditLogPublisher) {
+                          final NakadiAuditLogPublisher auditLogPublisher,
+                          final TransactionTemplate transactionTemplate,
+                          final Environment environment) {
         this.objectMapper = objectMapper;
         this.storageDbRepository = storageDbRepository;
-        this.defaultStorage = defaultStorage;
-        this.curator = zooKeeperHolder.get();
         this.featureToggleService = featureToggleService;
         this.auditLogPublisher = auditLogPublisher;
-        this.zkNotificationThread = Executors.newSingleThreadScheduledExecutor();
+        this.transactionTemplate = transactionTemplate;
+        this.environment = environment;
     }
 
-    private void refreshDefaultStorageAndCreateWatcher() {
-        final byte[] defaultStorageId;
-        try {
-            defaultStorageId = curator.getData()
-                    .usingWatcher(
-                            (Watcher) evt -> zkNotificationThread.submit(this::refreshDefaultStorageAndCreateWatcher))
-                    .forPath(ZK_TIMELINES_DEFAULT_STORAGE);
-        } catch (Exception ex) {
-            LOG.error("Failed to refresh default storage id, updates may be blocked", ex);
-            zkNotificationThread.schedule(this::refreshDefaultStorageAndCreateWatcher, 10, TimeUnit.SECONDS);
+    // TODO: the next method is completely useless in the real world, and makes sense only in test environment.
```

Comment:
Suggestion: refer to the method by name rather than saying next method.
And it is better to have an action item in mind if we mark it with `TODO`, otherwise, this could be an informational comment without the `TODO`.

Reply:
You are probably right, we wont fix this problem at all. I will remove the comment.